### ISSUE TITLE
Al excel de ranking le faltan algunos valores

### ISSUE THREAD

**Issue by emartinm** [emartinm/lsql#55]
El Excel generado parece tener los datos desplazados en algunas columnas, y no muestra la puntuación, el nombre de usuario (esto es particularmente importante) ni el número de resueltos. Adjunto un ZIP que contienen el HTML con el ranking y el Excel resultante, que debería ser una copia exacta.

¿@Tamiflu9 , podrías ver qué puede estar pasando? El test de unidad tenía éxito, pero no sé si era demasiado pequeño.

[ejemplo.zip](https://github.com/emartinm/lsql/files/6216118/ejemplo.zip)

**Comment by Tamiflu9** [comment]
Si el problema es por los logros, que no se porque en los alumnos con algún logro no detecta el nombre y los coge como `None`. Y como tengo puesto que no escriba nada cuando no detecta el nombre de alguna etiqueta, por eso las celdas están desplazadas.

**Comment by emartinm** [comment]
En el Excel que adjunto creo que no aparece ningún nombre de usuario, tenga logros o no. Y tampoco aparecen la puntuación ni el número de problemas resueltos. Creo que lo mejor es que os coordinéis @Tamiflu9  @Dashito14 para arreglar esta interacción entre los logros y la descarga del ranking.

**Comment by Dashito14** [comment]
El problema está en los logros. Al coger por etiqueta `<td>` se lía con el trofeo y el x2, x3... Algo que se me ha ocurrido es hacer como lo mostré al principio. Una columna nueva para los logros en el que pongamos el trofeo y el número de logros, o incluso al ser una columna nueva podríamos intentar hacer que se muestren el número de copas de logros que tiene conseguidos, como si fuera un militar, como dijiste.

**Comment by emartinm** [comment]
Yo estoy bastante contento con la presentación actual, así que preferiría no cambiar cómo se visualizan. Para facilitar extraer únicamente el nombre, creo que la mejor opción es estructurar la casilla de nombre de usuario con dos `span`:

    <td class="border border-dark">
        <span class="ranking-username">nombre_usuario</span>
        <span class="ranking-achievement">
            <a href="/sql/achievements/13" class="gold"><i class="fas fa-trophy gold"></i>x4</a>
        </span>
    </td>

Así, en la celda de usuarios únicamente hay que acceder al span de clase `ranking-username` y extraer su texto.

¿Esto también arreglaría que no aparezcan los puntos ni el número de problemas resuelto?

**Comment by emartinm** [comment]
Sobre el papel, los `span` no introducen saltos de línea ni afectan a la presentación

**Comment by Dashito14** [comment]
En un principio si podemos acceder solo a uno de los `span` sí que lo solucionará. El problema de no mostrar ni puntos ni problemas es que se mostraban una celda a la izquierda si el usuario tenía un logro completado ya que no mostraba el nombre, ahora lo probamos.

**Comment by emartinm** [comment]
Siento volver con esto, pero acabo de subirlo al servidor y siguen sin aparecer la puntuación ni los resueltos, ver 
[ranking.xlsx](https://github.com/emartinm/lsql/files/6224249/ranking.xlsx). Los nombres de usuario ya sí que se muestran sin problemas. El código HTML del ranking es este: [ranking.txt](https://github.com/emartinm/lsql/files/6224267/ranking.txt)

¿Qué más puede estar pasando? @Dashito14 @Tamiflu9

**Comment by Tamiflu9** [comment]
Exactamente en que fila no aparecen los resultados y la puntuación? que estoy comparando los resultado de la tabla del ranking con el excel que me has pasado y no veo donde está la diferencia.

**Comment by Dashito14** [comment]
No veo yo tampoco que datos faltan

**Comment by Dashito14** [comment]
![image](https://user-images.githubusercontent.com/48255774/112949727-d2c19f00-9139-11eb-8d03-d851880421c2.png)
![image](https://user-images.githubusercontent.com/48255774/112949785-e40aab80-9139-11eb-8abf-8f7c5ab954c8.png)

No sé qué datos te faltan, yo veo todo de una imagen a la otra.

**Comment by emartinm** [comment]
Ya he descubierto el misterio: esas celdas tienen saltos de línea y espacios y Excel los quita mientras que LibreOffice no. Y como el valor está en la segunda línea, en la vista normal no lo ves:

![Captura de pantalla de 2021-03-30 11-17-44](https://user-images.githubusercontent.com/1685343/112965669-c47b7f00-9149-11eb-83aa-c03f4ec6737f.png)

Voy a poner unos strip() en el código para quitar todos los espacios en blanco en ambos extremos.

**Comment by Dashito14** [comment]
Anda, que es con el programa que lo abres. ¿Has podido solucionarlo con los `strip()`?

**Comment by emartinm** [comment]
Sí, acabo de añadir los strip en todas las escrituras de celda y parece que funciona en mi ejemplo local. Ahora lo despliego en el servidor y lo pruebo con los Excel más grandes, pero estoy casi seguro que era eso (la celda HTML con los datos tenía saltos de línea y espacios, que al usar plantillas es **muy común**, y al coger el texto tal cual se metía todo eso en la hoja de cálculo).

Pero idealmente estará ya resuelto en el commit 8570561e14596827acb6847c43ec0ee7364500f4
